### PR TITLE
fix gh action

### PR DESCRIPTION
**Pull request by btc-fan**
## Reasoning behind the pull request
- Addressed issues where the workflow was incorrectly using the target branch instead of the PR source branch.
- Fixed branch resolution logic to correctly handle cases where branches are missing or comments don't specify branches.
- Resolved dependency conflicts in mx-chain-testing-suite related to protobuf versions.
  
## Proposed changes
- Updated GitHub Action to correctly fetch and use the latest commit from the PR source branch.
- Implemented logic to skip branch overrides when the last comment does not contain "Run Tests:".
- Adjusted requirements.txt in mx-chain-testing-suite to resolve protobuf version conflicts.

## Testing procedure
- Verified that the workflow correctly checks out the source branch and uses the latest commit hash.
- Tested the branch override logic by adding and omitting "Run Tests:" in PR comments.
- Confirmed that the dependency update does not introduce new issues and that all tests pass.

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Comment by btc-fan** [comment]
Run Tests:
mx-chain-simulator-go: main
mx-chain-testing-suite: main

**Comment by btc-fan** [comment]
bla bla

**Comment by btc-fan** [comment]
Run Tests:
mx-chain-simulator-go: main
mx-chain-testing-suite: main

**Comment by btc-fan** [comment]
dsfgdsfyg dsfg dsfg

**Comment by btc-fan** [comment]
Run Tests:
mx-chain-simulator-go: main
mx-chain-testing-suite: main

**Comment by btc-fan** [comment]
Run Tests:
mx-chain-simulator-go: main
mx-chain-testing-suite: main

**Comment by btc-fan** [comment]
Run Tests:
mx-chain-simulator-go: main
mx-chain-testing-suite: main

**Comment by btc-fan** [comment]
zsdgvfdsfgsd

File: .github/workflows/build_and_run_chain_simulator_and_execute_system_test.yml [hunk 1]
```diff
@@ -35,48 +35,68 @@ jobs:
...
+
+            if (lastComment && lastComment.body.includes('Run Tests:')) {
+              const body = lastComment.body.trim();
+              core.setOutput('latest_comment', body);
 
-            if (latestComment) {
-              core.setOutput('latest_comment', latestComment.body);
+              // Parse the branches from the last comment
+              const simulatorBranchMatch = body.match(/mx-chain-simulator-go:\s*(\S+)/);
+              const testingSuiteBranchMatch = body.match(/mx-chain-testing-suite:\s*(\S+)/);
```
Comment (andreibancioiu) [inline .github/workflows/build_and_run_chain_simulator_and_execute_system_test.yml:80]: :see_no_evil: 

GH workflow inputs (pretty visual forms) would have permitted us to not write regex.

GH workflow inputs support default values (of course, without additional conditions as above, but might be sufficient).

**Review by andreibancioiu — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Pull request merged**